### PR TITLE
Fix: Pass scheduler_specific_kwargs to inverse_sqrt scheduler

### PR DESCRIPTION
**Pull request by r266-tech**
## What does this PR do?

Fixes #44908

The `get_inverse_sqrt_schedule` function accepts `timescale` and `last_epoch` parameters, but `get_scheduler` was not forwarding `scheduler_specific_kwargs` to it. This caused user-provided kwargs like `timescale` to be silently ignored.

## Before/After

**Before:**
```python
if name == SchedulerType.INVERSE_SQRT:
    return schedule_func(optimizer, num_warmup_steps=num_warmup_steps)
```

**After:**
```python
if name == SchedulerType.INVERSE_SQRT:
    return schedule_func(optimizer, num_warmup_steps=num_warmup_steps, **scheduler_specific_kwargs)
```

## How to reproduce

Trigger a training job with `lr_scheduler_type="inverse_sqrt"` and set `lr_scheduler_kwargs={"timescale": 100}`. Before this fix, `timescale` is silently ignored and defaults to `num_warmup_steps`. After this fix, the user-provided `timescale` takes effect.

### CODE REVIEW

**Comment by Rocketknight1** [comment]
Duplicate of #44909 **and** it looks like a pure code agent PR. Please check for other PRs before opening your own!
